### ISSUE TITLE
-sorted support for getfasta (willing to collaborate)

### ISSUE THREAD

**Issue by bmorledge-hampton19** [arq5x/bedtools2#1050]
In my experience with using bedtools, I have noticed that the getfasta command is commonly a bottleneck in my pipelines, despite the fact that the task does not seem particularly complex. I am wondering if this tool could be sped up by including an alternative algorithm for sorted input, similar to how the `-sorted` option functions in bedtools intersect. I'm a bit rusty reading C/C++ code, but it looks like the current execution acquires sequences independently from one another and indexing only helps acquire the starting point for each fasta header. In the case that many sequences need to be retrieved from the end of a large chromosome, this appears to be extremely suboptimal.

I'm willing to help implement an algorithm that expects sorted (`-k1,1 -k2,2n`) bed input and thus only has to iterate through the fasta file once to obtain all the relevant sequences with minimal memory usage. Currently, it looks like the algorithm is heavily dependent on htslib, which is admittedly a bit opaque to me. I'm not sure whether an algorithm that leverages sorted input is compatible with htslib, and I'm not sure how much functionality would have to be redesigned without the use of this library. Is there anyone with more experience in the source who can weigh in on the potential usefulness of this algorithm? I'm happy to do most of the heavy lifting here, but I want to make sure I'm approaching the task well-informed.

tldr: It seems that an algorithm expecting sorted input could substantially improve the getfasta command. I'm happy to help implement such an algorithm, but want to make sure the task is tractable/reasonable first.
